GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -657,11 +657,10 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
                     // additional work is needed for any files\folders inside the folder being moved)
                     if (result == FileSystemTaskResult.Success && !string.IsNullOrEmpty(gitUpdate.VirtualPath))
                     {
+                        this.AddToNewlyCreatedList(gitUpdate.VirtualPath, isFolder: true);
                         result = this.TryAddModifiedPath(gitUpdate.VirtualPath, isFolder: true);
                         if (result == FileSystemTaskResult.Success)
                         {
-                            this.newlyCreatedFileAndFolderPaths.Add(gitUpdate.VirtualPath);

```

Comment:
Since the code now checks the modified paths to be able to add to the newly created list, this code needed to move to before the path is added to modified paths otherwise it would never add to the newly created list.